theano/gpuarray/opt.py

```diff
@@ -756,6 +756,11 @@ def max_inputs_to_GpuElemwise(node):
 optdb.register('gpua_inplace_opt', inplace_gpu_elemwise_opt, 75,
                'inplace_elemwise_optimizer', 'fast_run', 'inplace', 'gpuarray')
 
+gpu_elemwise_alloc = gof.local_optimizer([GpuElemwise])(
+    tensor.opt.local_elemwise_alloc_op(GpuElemwise, GpuAlloc, GpuDimShuffle)
+)
+
+tensor.opt.register_specialize_device(gpu_elemwise_alloc)
```

Comment:
As far as I know, it transforms elemwise(alloc(), other) into elemwise(dimshuffle(), other) if the shapes allow it.

It's not required per se, but I sense it might make things go faster.